components/bms_worker/src/CAN/CANIO-tx.c

```diff
@@ -99,78 +108,68 @@ static const packTable_S MSG_packTable_1Hz[] = {
  */
 void CANTX_BUS_A_SWI(void)
 {
-    // TODO: add overrun detection here
-    if (cantx.tx_100Hz_msg == MSG_packTable_100Hz_SIZE)
+    if (cantx.tx_100Hz_msg != MSG_packTable_100Hz_SIZE)
     {
-        goto continue1;
-    }
+        static uint8_t counter_100Hz = 0U;
+        CAN_data_T     message_100Hz = { 0U };
```

Comment:
do you really need to have a distinct message object for each frequency? You send each message immediately after packing, so I think it should be fine to just reuse the same object but `memset` it to 0 before the next pack function call?